runtime/bcutil/cfreader.c

```diff
@@ -2643,8 +2643,14 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 	/* Make sure the structure and static verification below uses the class file version
 	 * number. VM version is maintained in vmVersionShifted.
 	 */
-	flags &= ~BCT_MajorClassFileVersionMask;
-	flags |= ((UDATA) classfile->majorVersion) << BCT_MajorClassFileVersionMaskShift;
+	if (J9_ARE_ANY_BITS_SET(findClassFlags, J9_FINDCLASS_FLAG_UNSAFE)
+	&& ((vmVersionShifted == BCT_Java8MajorVersionShifted) || (vmVersionShifted == BCT_Java11MajorVersionShifted))
+	) {
+		/* Ignore the check for Unsafe.defineClass() on Java 8 & Java 11 */
+	} else {
+		flags &= ~BCT_MajorClassFileVersionMask;
+		flags |= ((UDATA) classfile->majorVersion) << BCT_MajorClassFileVersionMaskShift;
+	}
```

Comment:
I don't understand this change.  What check later in the code are we trying to avoid by this?

Reply:
The late check on `invokestatic` in the case of unsafe needs to avoided in `checkBytecodeStructure` at  the code at `/runtime/bcverify/staticverify.c`:
```
checkBytecodeStructure (J9CfrClassFile * classfile, UDATA methodIndex, UDATA length, U_8 * map, J9CfrError * error, U_32 flags, I_32 *hasRET)
{
...
		case CFR_BC_invokestatic:
			...
			if (info->tag != CFR_CONSTANT_Methodref) {
------------------->    if (((flags & BCT_MajorClassFileVersionMask) >= BCT_Java8MajorVersionShifted)
				&& (bc != CFR_BC_invokevirtual) && (info->tag == CFR_CONSTANT_InterfaceMethodref)
				) {
           ...
					/* Valid - take no action */
				} else {
   ---------------------->errorType = J9NLS_CFR_ERR_BC_NOT_METHODREF__ID; 
					/* Jazz 82615: Set the constant pool index to show up in the error message framework */
					errorDataIndex = index;
					goto _verifyError;
				}
			}
```
We have to disable the check for unsafe to address the test failure with `Unsafe.defineClass()` detected in the OpenJ9 test suite `jsr335_interfacePrivateMethod` in personal builds (Java 8 & 11) as I mentioned at https://github.com/eclipse/openj9/issues/9880#issuecomment-649677295
```
java.lang.VerifyError: JVMCFRE116 invoke bytecode must reference a Methodref; class=jdk/internal/reflect/GeneratedMethodAccessor19, method=invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;, pc=20
Exception Details:
  Location:
---> jdk/internal/reflect/GeneratedMethodAccessor19.invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object; 
@20: JBinvokestatic
  Reason:
    Wrong type at the index 10 of constant pool
  Exception Handler Table:
    bci [0, 20] => handler: 24
    bci [0, 20] => handler: 24
    bci [20, 23] => handler: 36
	at java.base/jdk.internal.misc.Unsafe.defineClass(Unsafe.java:1510) <--------
	at java.base/jdk.internal.reflect.ClassDefiner.defineClass(ClassDefiner.java:63)
	at java.base/jdk.internal.reflect.MethodAccessorGenerator$1.run(MethodAccessorGenerator.java:400)
	at java.base/jdk.internal.reflect.MethodAccessorGenerator$1.run(MethodAccessorGenerator.java:394)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:678)
	at java.base/jdk.internal.reflect.MethodAccessorGenerator.generate(MethodAccessorGenerator.java:393)
	at java.base/jdk.internal.reflect.MethodAccessorGenerator.generateMethod(MethodAccessorGenerator.java:75)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:53)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.getLookupObject(Test_ReflectionAndMethodHandles.java:62)
	at org.openj9.test.jsr335.interfacePrivateMethod.Test_ReflectionAndMethodHandles.test_unreflect_child_public_static_m
```

Reply:
I changed the code to (removing the check on Java 8 & 11)
```
	if (J9_ARE_NO_BITS_SET(findClassFlags, J9_FINDCLASS_FLAG_UNSAFE)) {
	flags |= ((UDATA) classfile->majorVersion) << BCT_MajorClassFileVersionMaskShift;
		flags &= ~BCT_MajorClassFileVersionMask;
		flags |= ((UDATA) classfile->majorVersion) << BCT_MajorClassFileVersionMaskShift;
	}
```
to see how it goes in personal builds on Java 8 & 11.